src/net_processing.cpp

```diff
@@ -3265,10 +3271,32 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
 
         if (best_block != nullptr) {
-            if (MaybeSendGetHeaders(pfrom, m_chainman.ActiveChain().GetLocator(m_chainman.m_best_header), *peer)) {
-                LogPrint(BCLog::NET, "getheaders (%d) %s to peer=%d\n",
-                        m_chainman.m_best_header->nHeight, best_block->ToString(),
-                        pfrom.GetId());
+            // If we haven't started initial headers-sync with this peer, then
+            // consider sending a getheaders now. On initial startup, there's a
+            // reliability vs bandwidth tradeoff, where we only trying doing
+            // initial headers sync with one peer at a time, with a long
+            // timeout (at which point, if the sync hasn't completed, we will
+            // disconnect the peer and then choose another).  In the meantime,
+            // as new blocks are found, we are willing to add one new peer per
```

Comment:
This seems a corner case that blocks are announced to us via inv, no? So I am wondering why such a new block is a good signal to add the peer as a initial headers sync peer, as opposed to, let's say, a random timer.

Reply:
In that case, could make sense to correct the doc, which says "If a node fell back to sending blocks by inv,
                    // it's probably for a re-org."

Reply:
I guess words like "probably" are confusing in comments, as there is context that is missing.  I'll update both comments.

Reply:
Improved the comments and removed the "probably for a re-org" language.